v-next/hardhat-utils/src/sub-process.ts

```diff
@@ -0,0 +1,27 @@
+import { spawn } from "node:child_process";
+
+/**
+ * Spawns a detached subprocess to execute a given file with optional arguments.
+ *
+ * @param absolutePathToSubProcessFile - The absolute path to the subprocess file to be executed.
+ * @param [args=[]] - Optional arguments to pass to the subprocess.
+ *
+ * The subprocess runs in a detached mode and its standard input/output is ignored.
+ * This function does not wait for the subprocess to complete and the subprocess is unreferenced
+ * to allow the parent process to exit independently.
+ */
+export function spawnDetachedSubProcess(
+  absolutePathToSubProcessFile: string,
+  args: string[] = [],
+): void {
+  const subprocess = spawn(
+    process.execPath,
+    ["--import", "tsx/esm", absolutePathToSubProcessFile, ...args],
```

Comment:
Let's only add this import if the file is `.ts`.

Reply:
Done [here](https://github.com/NomicFoundation/hardhat/pull/5471/commits/9fa9f2c7ac30b7ffbd5ea007eec221fbf616a73c)